assignments/lambda-classes.js

```diff
@@ -1 +1,175 @@
 // CODE here for your Lambda Classes
+class Person { //This is the Class that all other classes will extend from - Ben Allen
+    constructor(attributes){
+        this.name = attributes.name;
+        this.age = attributes.age;
+        this.location = attributes.location;
+    }
+    speak() { //Method
+        console.log(`Hello my name is ${this.name}, I am from ${this.location}.`);
+    }
+}
+
+class Insructor extends Person {
+    constructor(attributes){
+        super(attributes);
+        this.specialty = attributes.specialty;
+        this.favLanguage = attributes.favLanguage;
+        this.catchPhrase = attributes.catchPhrase;
+    }
+    demo(subject) { //Method
+        console.log(`Today we are learning about ${subject}.`);
+    }
+    
+    grade(studentObj, subject) { //Method
+        console.log(`${studentObj.name} receives a perfect score on ${subject}`);
+    }
+    
+}
+
+class Student extends Person {
+    constructor(attributes){
+        super(attributes);
+        this.previousBackground = attributes.previousBackground;
+        this.className = attributes.className;
+        this.favSubjectsArray = attributes.favSubjectsArray;
+    }
+    listsSubject(){ //Method
+        console.log('My favorite subjects are:');
+        for(let i = 0; i < this.favSubjectsArray.length; i++){
+            console.log(this.favSubjectsArray[i]);
+        }
+    }
+    PRAssignment(subject) { //Method
+        console.log(`${this.name} has submitted a PR for ${subject}.`);
+    }
+    sprintChallenge(subject) { //Method
+        console.log(`${this.name} has begun sprint challenge on ${subject}.`);
+    }
+}
+
+class ProjectManager extends Person {
+    constructor(attributes) {
+        super(attributes);
+        this.gradClassName = attributes.gradClassName;
+        this.favInstructor = attributes.favInstructor;
+    }
+    standUp(channel) { //Method
+        console.log(`${this.name} announces to ${channel}, @channel stand up time!`);
+    }
+    debugsCode(studentObj, subject) { //Method
+    console.log(`${this.name} debugs ${studentObj.name}'s code on ${subject}`);
+    }
+}
+console.log(`*************Students***************`)
+//New Students
+const benAllen = new Student({
```

Comment:
Great use of variable names to differentiate your constructors from your instantiated objects.